src/Protocol/Feed.php

```diff
@@ -1108,9 +1109,8 @@ private static function entryFooter(DOMDocument $doc, $entry, array $item, array
 
 		if ($item['gravity'] != GRAVITY_PARENT) {
 			$parent = Item::selectFirst(['guid', 'author-link', 'owner-link'], ['id' => $item['parent']]);
-			$parent_item = (($item['thr-parent']) ? $item['thr-parent'] : $item['parent-uri']);
 
-			$thrparent = Item::selectFirst(['guid', 'author-link', 'owner-link', 'plink'], ['uid' => $owner["uid"], 'uri' => $parent_item]);
+			$thrparent = Item::selectFirst(['guid', 'author-link', 'owner-link', 'plink'], ['uid' => $owner["uid"], 'uri' => $item['thr-parent']]);
```

Comment:
On really old posts the field `thr-parent` had been empty. So I guess we need an update job that fills `thr-parent` with the value of `parent-uri` when it's empty.

Reply:
I just checked. Indeed I have got around 16,000 posts from the year 2012 where this is the case.